var/spack/repos/builtin/packages/py-genshi/package.py

```diff
@@ -8,11 +8,27 @@
 
 class PyGenshi(PythonPackage):
     """Python toolkit for generation of output for the web"""
-    homepage = "https://genshi.edgewall.org/"
-    url      = "http://ftp.edgewall.com/pub/genshi/Genshi-0.7.tar.gz"
+    pypi = "Genshi/Genshi-0.7.7.tar.gz"
 
+    version('0.7.7', sha256='53920c8ed890a3ce0e274845abe82d34736191ffc51300788d4cc71e4fcdd0eb')
     version('0.7', sha256='1d154402e68bc444a55bcac101f96cb4e59373100cc7a2da07fbf3e5cc5d7352')
     version('0.6.1', sha256='fed947f11dbcb6792bb7161701ec3b9804055ad68c8af0ab4f0f9b25e9a18dbd')
     version('0.6', sha256='32aaf76a03f88efa04143bf80700399e6d84eead818fdd19d763fd76af972a4b')
```

Comment:
You'll need to update the checksums, PyPI tarballs contain fewer files.

Reply:
I did run `spack checksum py-genshi` in that commit.

Just now I commented out those older versions and tried again, but the checksums are the same as pypi for whatever reason:

```console
$ spack checksum py-genshi  0.6 0.6.1 0.7
==> Found 3 versions of py-genshi:
  
  0.7    https://files.pythonhosted.org/packages/source/G/Genshi/Genshi-0.7.tar.gz
  0.6.1  https://files.pythonhosted.org/packages/source/G/Genshi/Genshi-0.6.1.tar.gz
  0.6    https://files.pythonhosted.org/packages/source/G/Genshi/Genshi-0.6.tar.gz

==> Fetching https://files.pythonhosted.org/packages/source/G/Genshi/Genshi-0.7.tar.gz
==> Fetching https://files.pythonhosted.org/packages/source/G/Genshi/Genshi-0.6.1.tar.gz
==> Fetching https://files.pythonhosted.org/packages/source/G/Genshi/Genshi-0.6.tar.gz

    version('0.7',   sha256='1d154402e68bc444a55bcac101f96cb4e59373100cc7a2da07fbf3e5cc5d7352')
    version('0.6.1', sha256='fed947f11dbcb6792bb7161701ec3b9804055ad68c8af0ab4f0f9b25e9a18dbd')
    version('0.6',   sha256='32aaf76a03f88efa04143bf80700399e6d84eead818fdd19d763fd76af972a4b')
```

Reply:
Oh whoops, yes the checksum for 0.7.7 is different!

Reply:
I got bitten by https://github.com/spack/spack/issues/30864 and need to rebase to use that fix.

```console
$ spack spec -I ome-files-cpp@master %gcc@11.3.0
==> Error: No valid version for 'python' satisfies '@3:'
```

Reply:
^ @alalazo @becker33 @scheibelp

Reply:
@adamjstewart I think I've fixed everything from your comments.  Thanks again.